### PR TITLE
 Source Hubspot: new stream Email Subscriptions

### LINKED ISSUE #17034: Source Hubspot: new stream Email Subscriptions
## Tell us about the problem you're trying to solve
[Email Subscriptions](https://legacydocs.hubspot.com/docs/methods/email/email_subscriptions_overview)
The Email Subscriptions API can be used to build integrations with other Business Intelligence (BI) systems for reporting.

[Scopes](https://legacydocs.hubspot.com/docs/methods/oauth2/initiate-oauth-integration#)
Content scope provides access to _All CMS APIs, Calendar API, Email and Email Events APIs_. Account types that can access the scope - _Marketing Professional_, or _Enterprise_.

## Describe the solution you’d like
Research pros and cons for implementing this stream.
_Research should contain: what stream should be implemented, permissions and scopes (if needed), paid plan (if needed)._

### PR DESCRIPTION
**Pull request by aazam-gh**
## What
Adding a new stream email subscriptions
Returns all email subscription types that have been created in the given Hub ID.

## How
The connector adds the email subscription API stream
Fixes #17034 

## Recommended reading order
1. `source.py`
2. `streams.py`

## 🚨 User Impact 🚨
No changes to existing code

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

image

### CODE REVIEW

**Comment by Alcadeus0** [comment]
image
The integration test for incremental read fails. The stream is not an incremental one

**Comment by Alcadeus0** [comment]
@YowanR can you take a look at this?

**Comment by YowanR** [comment]
@Alcadeus0 Is it possible to make this stream incremental? Is there a cursor available?

**Comment by Alcadeus0** [comment]
@YowanR Unfortunately this stream doesn't have a cursor field available

**Comment by YowanR** [comment]
@sh4sh @natalyjazzviolin Can you please take a look at this PR? Thanks 🙏

**Comment by Alcadeus0** [comment]
image

**Comment by Alcadeus0** [comment]
> /test connector=connectors/source-hubspot
> > :clock2: connectors/source-hubspot https://github.com/airbytehq/airbyte/actions/runs/4481449916

What does that mean?

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/schemas/email_subscriptions.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": ["null", "object"],
+    "properties": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/schemas/email_subscriptions.json:4]: missing the additionalProperties field, please check others stream.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments about the stream schema please fix them.

**Comment by Alcadeus0** [comment]
@marcosmarxm Are the tests not successful? It should have merged right?

**Comment by marcosmarxm** [comment]
> @marcosmarxm Are the tests not successful? It should have merged right?

Yes, Tests are working now. I need the final approval from the connector team and it is ready to merge. There isn't anything in your side right now! Thanks for the contribution @Alcadeus0

**Review by marcosmarxm — APPROVED** [review]
Thanks @Alcadeus0

**Comment by Alcadeus0** [comment]
Thanks for your help with the review :)

**Review by lazebnyi — APPROVED** [review]
Nice!

**Pull request merged**